Dockerfile.rp-full-dev

```diff
@@ -0,0 +1,23 @@
+FROM registry.access.redhat.com/ubi8/toolbox:8.10
+
+# Needed for installing packages
+USER root 
+
+RUN dnf install https://packages.microsoft.com/config/rhel/8/packages-microsoft-prod.rpm -y \
+    && dnf install make golang azure-cli -y \
+    && dnf install https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm -y \
+    && dnf install screen openvpn -y \
+    && dnf install jq docker device-mapper-devel skopeo -y \
```

Comment:
I have a WIP PR to address this naming confusion - but `podman` and `podman-docker` should be sufficient and preferred over `docker`.

Reply:
I tried using Podman instead of Docker, but it introduced https://github.com/Azure/ARO-RP/pull/3764#issuecomment-2304826079 issue. I would prefer to address that in a follow-up. Reviewing the PR is already quite complex.